fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java

```diff
@@ -124,10 +125,19 @@ private void auditAfterExec(String origStmt, StatementBase parsedStmt, Data.PQue
 
         if (ctx.getState().isQuery()) {
             MetricRepo.COUNTER_QUERY_ALL.increase(1L);
-            if (ctx.getState().getStateType() == QueryState.MysqlStateType.ERR
-                    && ctx.getState().getErrType() != QueryState.ErrType.ANALYSIS_ERR) {
-                // err query
-                MetricRepo.COUNTER_QUERY_ERR.increase(1L);
+            if (ctx.getState().getStateType() == QueryState.MysqlStateType.ERR) {
+                InternalErrorCode errorCode = ctx.getState().getInternalErrorCode();
+                if (errorCode == InternalErrorCode.INTERNAL_ERR) {
+                    // err query
+                    MetricRepo.COUNTER_QUERY_ERR.increase(1L);
+                } else {
+                    MetricRepo.COUNTER_QUERY_ERR_OTHER.increase(1L);
+                    MetricRepo.HISTO_QUERY_LATENCY.update(elapseMs);
```

Comment:
Why update `HISTO_QUERY_LATENCY` for this case?

Reply:
Userexception (timeout, cancel, etc) will be treated as `ANALYSIS_ERR`，`ANALYSIS_ERR` will be recorded in the query time. It is understandable that a known error query will be recorded in the query time.
Therefore, this PR still records known errors (timeout, cancel, etc.) in the query time, and other errors (RPC, unknown errors) are not recorded in the query time.

Reply:
Maybe we can remove this logic and only record the successful query in the query time